components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -48,7 +49,25 @@ public protocol PersistCallback {
 fileprivate let queue = DispatchQueue(label: "com.mozilla.firefox-account")
 
 open class FirefoxAccount: RustHandle {
-    fileprivate static var persistCallback: PersistCallback?
+    fileprivate var persistCallback: PersistCallback?
+
+    func tryPersistState() {
+        queue.async {
+            guard let cb = self.persistCallback else {
+                return
+            }
+            do {
+                let json = try self.toJSONInternal()
+                DispatchQueue.global(qos: .background).async {
```

Comment:
Should we be the ones deciding this? Or should the callback. More of a question for @garvankeeley also.

Reply:
Good question. In Swift I/O APIs (networking and storage) callbacks are by-convention off-main thread. Because the app services stuff is an I/O API it is correct (by de-facto standards) to callback on the background queue.